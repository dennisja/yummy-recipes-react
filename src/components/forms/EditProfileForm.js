import React, { Component } from 'react';
import User from '../../helpers/User';
import Token from '../../helpers/Token';
import {displayError} from '../Utilities';

export default class EditProfileForm extends Component {
    state = {
        firstname: "",
        lastname: "",
        email: "",
    }

    componentDidMount(){
        // get data of the logged in user
        const {firstname, lastname, email} = Token.getUserData()
        // set state to current values of user data
        this.setState({
            firstname,
            lastname,
            email
        })
    }

    handleInputChange = (event)=>{
        const target = event.target;
        this.setState({
            [target.name] : target.value
        });
    }

    handleFormSubmit = (event)=>{
        event.preventDefault();
        // make request to update user information
        // if you add another value of state that is not among the sent data
        // please update the sent data
        User.editUserDetails(this.state)
        .then(response=>{
            window.Materialize.toast(response.data.message, 4000)
        })
        .catch(error=>{
            displayError(error)
        })
    }

    render() {
        return (
          <form onSubmit={this.handleFormSubmit}>
            <div className="input-field">
              <label htmlFor="firstname">Lastname</label>
              <input type="text" name="firstname" id="firstname" onChange={this.handleInputChange} value={this.state.firstname} required />
            </div>
            <div className="input-field">
              <label htmlFor="lastname">Lastname</label>
              <input type="text" name="lastname" id="lastname" onChange={this.handleInputChange} value={this.state.lastname} required />
            </div>
            <div className="input-field">
              <label htmlFor="email">Email</label>
              <input type="email" name="email" id="email" onChange={this.handleInputChange} value={this.state.email} required />
            </div>
            <div className="center-align">
              <input type="submit" value="Edit Profile" id="edit_profile" name="edit_profile" className="btn btn-small orange" />
            </div>
          </form>
        );
    }
}
