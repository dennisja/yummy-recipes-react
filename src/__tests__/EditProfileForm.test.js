import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditProfileForm from '../components/forms/EditProfileForm';

describe('EditProfileForm component', () => {
    localStorage.setItem('YUMMY_USER', { data: { firstname: 'd', lastname: 'e', email: 'mockEmail' } });
    const formWrapper = shallow(<EditProfileForm />);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(formWrapper)).toMatchSnapshot();
    });

    it('renders a form element', () => {
        expect(formWrapper.find('form').length).toBe(1);
    });

    it('renders the required form fields', () => {
        expect(formWrapper.find('#firstname')).toHaveLength(1);
        expect(formWrapper.find('#lastname')).toHaveLength(1);
        expect(formWrapper.find('#email')).toHaveLength(1);
        expect(formWrapper.find('#edit_profile')).toHaveLength(1);
    });

    it('renders inputs with user details', () => {
        formWrapper.update();
        expect(formWrapper.find('#firstname').props().value).toBe('d');
        expect(formWrapper.find('#lastname').props().value).toBe('e');
        expect(formWrapper.find('#email').props().value).toBe('mockEmail');
    });

    it('submits data', () => {
        formWrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toBeCalled();
    });

    it('submit button submits the form', () => {
        formWrapper.find('#edit_profile').simulate('click');
        expect(preventDefault).toBeCalled();
    });
});

