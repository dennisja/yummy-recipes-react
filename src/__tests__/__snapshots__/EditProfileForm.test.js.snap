// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileForm component renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-field"
  >
    <label
      htmlFor="firstname"
    >
      Lastname
    </label>
    <input
      id="firstname"
      name="firstname"
      onChange={[Function]}
      required={true}
      type="text"
      value="d"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lastname"
    >
      Lastname
    </label>
    <input
      id="lastname"
      name="lastname"
      onChange={[Function]}
      required={true}
      type="text"
      value="e"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value="mockEmail"
    />
  </div>
  <div
    className="center-align"
  >
    <input
      className="btn btn-small orange"
      id="edit_profile"
      name="edit_profile"
      type="submit"
      value="Edit Profile"
    />
  </div>
</form>
`;
